raiden/raiden_service.py

```diff
@@ -539,6 +530,43 @@ def _start_transport(self, chain_state: ChainState) -> None:
             if neighbour != ConnectionManager.BOOTSTRAP_ADDR:
                 self.start_health_check_for(neighbour)
 
+    def _prepare_and_execute_alarm_first_run(self, last_log_block: BlockNumber) -> None:
+        """Prepares the alarm task callback and executes its first run
+
+        Complete the first_run of the alarm task and synchronize with the
+        blockchain since the last run.
+
+         Notes about setup order:
+         - The filters must be polled after the node state has been primed,
+           otherwise the state changes won't have effect.
+         - The alarm must complete its first run before the transport is started,
+           to reject messages for closed/settled channels.
+        """
+        assert not self.transport, f"Transport is running. node:{self!r}"
+
+        self.alarm.register_callback(self._callback_new_block)
+        self.alarm.first_run(last_log_block)
+        # The first run of the alarm task processes some state changes and may add
+        # new token network event filters when this is the first time Raiden runs.
+        # Here we poll for any new events that may exist after the addition of
+        # those event filters.
+        latest_block_num = self.chain.get_block(block_identifier="latest")["number"]
+        latest_confirmed_block_num = max(
+            GENESIS_BLOCK_NUMBER, latest_block_num - self.confirmation_blocks
+        )
+        latest_confirmed_block_num += 1
```

Comment:
Humm, this is getting one unconfirmed block, it shouldn't be necessary

Reply:
hahaha wtf? How did this get in there? Thanks for seeing it.